src/v/ssx/tests/work_queue_test.cc

```diff
@@ -146,4 +146,21 @@ TEST(WorkQueue, CanShutdownWithDelayedTasks) {
     EXPECT_FALSE(a2);
 }
 
+TEST(WorkQueue, RecursiveSubmit) {
```

Comment:
question: what would the result of this test have been prior to this change?

Reply:
It segfaulted in release mode only